src/components/timeline/TrackContextMenu.js

```diff
@@ -295,6 +295,24 @@ class TimelineTrackContextMenu extends PureComponent<Props> {
     return localTrackNames[rightClickedTrack.trackIndex];
   }
 
+  getRightClickedTrackType(rightClickedThreadIndex: TrackReference): string {
+    const { globalTrackNames, localTrackNamesByPid } = this.props;
+
+    if (rightClickedThreadIndex.type === 'global') {
+      return globalTrackNames[rightClickedThreadIndex.trackIndex];
+    }
+    const localTrackNames = localTrackNamesByPid.get(
+      rightClickedThreadIndex.pid
+    );
+    if (localTrackNames === undefined) {
+      console.error(
+        'Expected to find a local and globaltrack name for the given pid.'
+      );
+      return 'Unknown Track';
+    }
+    return localTrackNames[rightClickedThreadIndex.trackIndex];
```

Comment:
This is not getting the "track type" it's getting the "track name" instead. We should get the track type here. You should get the global tracks and local tracks from the redux state first. To be able to do that, you can use `getLocalTracksByPid` and `getGlobalTracks` selectors. You should use these inside the `mapStateToProps` and assign them to a prop.

Reply:
hello, @canova  could you please look at my function which I created for onClick.